Use parallel builds to speed up compilation

mherrmann opened the pull request:
Resolves #44.

mherrmann committed:
Use parallel builds to speed up compilation

mihaiplesa commented on build_omaha.py:
```diff
@@ -22,7 +23,8 @@ def build(omaha_dir, standalone_installers_dir, build_all):
     mode = 'all'
-  command = ['hammer.bat', 'MODE=' + mode, '--all', '--standalone_installers_dir=' + standalone_installers_dir]
+  command = ['hammer.bat', 'MODE=' + mode, '--all', '--standalone_installers_dir=' + standalone_installers_dir,
+             '-j%d' % multiprocessing.cpu_count()]
```
Do we want this to be passed from outside instead?

mherrmann replied:
I don't know for sure and would say the answer depends on whether this ability - to change the `-j` value from outside - will actually be used in practice. Or whether it's a general suggested pattern in the build to only ever take such values from outside. As it is, the advantage is that it's very simple.

atuchin-m commented:
Could we check this on a test build before merging?

mherrmann commented:
@atuchin-m: I kicked one off with these changes in https://github.com/brave/brave-core/pull/12557

simonhong reviewed (approved):
When I porked omaha to brave (maybe 4y ago),
I didn't use parallel build because I saw some build failure on my local machine with `-j` option.
Happy to see parallel build for omaha! :)

mherrmann commented:
@simonhong thank you for the review! My suspicion is that the build failures were/are due to missing Python dependency `pywin32`. See https://github.com/brave/devops/issues/7101.

mherrmann committed:
Log python.exe location from build script

This helps us debug why dependency pywin32 does not get picked up in CI.

mherrmann committed:
Fix the path of python.exe used in hammer.bat

Previously, Python was picked up from the PATH. This lead to confusion
when hammer.bat mentioned an install location of Python, but then
another copy from the PATH was used instead.

mherrmann committed:
Revert "Log python.exe location from build script"

This reverts commit 099786225d45b2ed52b95ad5bcce271f2d208784.
